intake_erddap/utils.py

```diff
@@ -11,3 +18,67 @@ def get_project_version() -> str:
     from intake_erddap import __version__
 
     return __version__
+
+
+def return_category_options(
+    server: str,
+    category: Optional[str] = "standard_name",
+) -> DataFrame:
+    """Find category options for ERDDAP server.
+
+    Parameters
+    ----------
+    server : str
+        ERDDAP server address, for example: "http://erddap.sensors.ioos.us/erddap"
+    category : str, optional
+        ERDDAP category for filtering results. Default is "standard_name" but another good option is
+        "variableName".
+
+    Returns
+    -------
+    DataFrame
+        Column "Category" contains all options for selected category on server. Column "URL" contains
+        the link for search results for searching for a given category value.
+    """
+
+    df = pd.read_csv(
+        f"{server}/categorize/{category}/index.csv?page=1&itemsPerPage=100000"
+    )
+
+    return df
+
+
+def match_key_to_category(
+    server: str,
+    key: str,
+    category: Optional[str] = "standard_name",
```

Comment:
Here, `category` is not an optional type because the default is a string literal.

```python
category: str = "standard_name"
```

Reply:
In the type annotations system, `Optional` means that the symbol can take on the value of `None`.

Reply:
Should be fixed now.